climakitae/threshold_tools.py

```diff
@@ -884,3 +889,523 @@ def return_period(ams):
+    """
+    Examples:
+        'Number of hours per year'
+        'Number of 4-hour events per 3-months'
+        'Number of days per year with conditions lasting at least 4-hours'
+    """
+
+    if exceedance_count.duration2 != exceedance_count.duration1:
+        dur_len, dur_type = exceedance_count.duration1
+        _s = "" if dur_len == 1 else "s"
+        dur_str = f" with conditions lasting at least {dur_len} {dur_type}{_s}"
+    else:
+        dur_str = ""
+
+    if exceedance_count.duration2 != exceedance_count.group:
+        grp_len, grp_type = exceedance_count.group
+        if grp_len == 1:
+            grp_str = f" each {grp_type}"
+        else:
+            grp_str = f" every {grp_len} {grp_type}s"
+    else:
+        grp_str = ""
+
+    per_len, per_type = exceedance_count.period
+    if per_len == 1:
+        period_str = f" each {per_type}"
+    else:
+        period_str = f" per {per_len}-{per_type} period"
+
+    _subtitle = _exceedance_count_name(exceedance_count)+ period_str + dur_str + grp_str
+    
+    return _subtitle
+
+#------------- Class and methods for the explore_exceedance GUI ---------------
+
+
+class ThresholdDataParams(param.Parameterized):
+    """
+    An object that holds the "Data Options" parameters for the 
+    explore.thresholds panel. 
+    """
+
+    # Define the params (before __init__ so that we can access them during __init__)
+    threshold_direction = param.ObjectSelector(default = "above", objects = ["above", "below"], label = "Direction")
+    threshold_value = param.Number(default = 0, label = "")
+    duration1_length = param.Integer(default = 1, bounds = (0, None), label = "")
+    duration1_type = param.ObjectSelector(default = "hour", objects = ["year", "month", "day", "hour"], label = "")
+    period_length = param.Integer(default = 1, bounds = (0, None), label = "")
+    period_type = param.ObjectSelector(default = "year", objects = ["year", "month", "day"], label = "")
+    group_length = param.Integer(default = 1, bounds = (0, None), label = "")
+    group_type = param.ObjectSelector(default = "hour", objects = ["year", "month", "day", "hour"], label = "")
+    duration2_length = param.Integer(default = 1, bounds = (0, None), label = "")
+    duration2_type = param.ObjectSelector(default = "hour", objects = ["year", "month", "day", "hour"], label = "")
+    smoothing = param.ObjectSelector(default="None", objects=["None", "Running mean"], label = "Smoothing")
+    num_timesteps = param.Integer(default=10, bounds=(0, None), label = "Number of timesteps")
+
+    units2 = param.ObjectSelector(objects=dict())
+    # variable2 = param.ObjectSelector(default="Air Temperature at 2m", objects=["Air Temperature at 2m"])
+    variable2 = param.ObjectSelector(default="Air Temperature at 2m", objects=dict())
+    cached_area2 = param.ObjectSelector(default="Sacramento County", objects=dict())
+    area_subset2 = param.ObjectSelector(default="CA counties", objects=["CA counties", "states"])
```

Comment:
I don't think the default should be Sacramento county (sorry, Owen & Beth). It should probably be CA state, since that will be what more users are interested in seeing.

Reply:
I think that this tool doesn't make much sense at a state level, because it uses an area average. I think it makes more sense for local analysis. But maybe I should just default to the first alphabetical county?